### DISCUSSION TITLE
How to define the OS Version of a bottle to `all:` in github actions?

### DISCUSSION THREAD

**Discussion by a4z** [Homebrew/discussions#4485] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
na, github ci question
```

### Output of `brew doctor`

```shell
na, github ci question
```

### Description of issue

For a bottle that is a header-only library and that has no arch or OS version dependency, 

How can I have the autogenerated github action configure so it's sets the bottle part to `sha256 cellar: :any_skip_relocation, all: ....` instead of using venture?

**Comment by SMillerDev** [comment]
If it detects that it is the same across all platforms it'll use all

**Reply by a4z** [reply]
hm, how could it detect that on a GitHub action? 
it clearly placed the OS Name in the bottle when it created the last release
https://github.com/a4z/homebrew-cpp-taps/releases

**Reply by ZhongRuoyu** [reply]
It's unlikely that a C++-based binary formula can get an `all` bottle because the binaries on macOS and Linux should be different.

However, if you want to make your bottle work on as many macOS versions as possible, I would recommend running CI on an old version of macOS (or, adding more macOS versions to the GitHub Actions matrix, i.e., building bottles for more macOS versions). A bottle built for an earlier macOS version can be used, if there is no bottle built for the current version.

At the time of writing, GitHub provides `macos-11`, `macos-12`, and `macos-13` runner images ([docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)). Note that GitHub-hosted Apple Silicon runners are not available yet, so ARM bottles can't be easily built.

**Reply by gromgit** [reply]
Actually, those are header-only bottles that differ only in metadata:
```
$ diff -ur strong_type.*
diff -ur strong_type.ventura/11/INSTALL_RECEIPT.json strong_type.x86_64_linux/11/INSTALL_RECEIPT.json
--- strong_type.ventura/11/INSTALL_RECEIPT.json	2023-05-06 02:04:40.000000000 +0800
+++ strong_type.x86_64_linux/11/INSTALL_RECEIPT.json	2023-05-06 02:07:32.000000000 +0800
@@ -16,7 +16,7 @@
   ],
   "time": null,
   "source_modified_time": 1683047841,
-  "compiler": "clang",
+  "compiler": "gcc-11",
   "aliases": [
 
   ],
@@ -36,11 +36,10 @@
   },
   "arch": "x86_64",
   "built_on": {
-    "os": "Macintosh",
-    "os_version": "macOS 13.3",
-    "cpu_family": "ivybridge",
-    "xcode": "14.2",
-    "clt": "14.3.0.0.1.1679647830",
-    "preferred_perl": "5.30"
+    "os": "Linux",
+    "os_version": "Ubuntu 22.04.2 LTS",
+    "cpu_family": "skylake",
+    "glibc_version": "2.35",
+    "oldest_cpu_family": "core2"
   }
 }
\ No newline at end of file
```

**Reply by a4z** [reply]
Yes, header-only libraries are just text files the user includes. So they can be the same over Arch and Mac OS Version
There are no binaries, will look at binary dependencies in a future step. 
For now, I plan to collect some header libs I need the most in one tap, so they exist.
For me, but others might find them also useful, XCode clang lags behind the C++ standard, and there are some header libs that add missing functionality that you have already on Linux gcc/clang and MS msvc/clang

**Reply by Idane** [reply]
What's the logic behind this? I have the same issue and I reduced the number of bottles to 1, still no all
